Doc/whatsnew/3.14.rst

```diff
@@ -117,6 +117,12 @@ Added support for converting any objects that have the
 :meth:`!as_integer_ratio` method to a :class:`~fractions.Fraction`.
 (Contributed by Serhiy Storchaka in :gh:`82017`.)
 
+io
+___
+
+* Reading text from a non-blocking stream with ``read`` may now raise a :exc:`BlockingIOError` if the operation cannot immediately return bytes.
+(Contributed by Giovanni Siragusa in :gh:`109523`.)
```

Comment:
```suggestion

io
--

* Reading text from a non-blocking stream with ``read`` may now raise a
  :exc:`BlockingIOError` if the operation cannot immediately return bytes.
  (Contributed by Giovanni Siragusa in :gh:`109523`.)
```

Reply:
FTR: we want two blank lines before the section "io" (each module is separated by two blank lines, but the contributions within a same module are separated by 1 blank line in general).